docs/core/knowledge-bases.rst

```diff
@@ -0,0 +1,416 @@
+      }
+
+  You can overwrite it by calling the function ``set_ordinal_mention_mapping``.
+  If you want to learn more about the usage of the mapping, go to section :ref:`resolve_mentions`.
+
+Defining the NLU Data
+---------------------
+
+To be able to understand that the user wants to retrieve some information from the knowledge base, you need to define
+a new intent, for example, ``query_knowledge_base``.
+The intent should contain all kind of user requests.
+
+Let's look at an example:
+
+.. code-block:: yaml
+
+    ## intent:query_knowledge_base
+    - what [restaurants](object_type:restaurant) can you recommend?
+    - list some [restaurants](object_type:restaurant)
+    - can you name some [restaurants](object_type:restaurant) please?
+    - can you show me some [restaurant](object_type:restaurant) options
+    - list [German](cuisine) [restaurants](object_type:restaurant)
+    - do you have any [mexican](cuisine) [restaurants](object_type:restaurant)?
+    - do you know the [price range](attribute:price-range) of [that one](mention)?
+    - what [cuisine](attribute) is it?
+    - do you know what [cuisine](attribute) the [last one](mention:LAST) has?
+    - does the [first one](mention:1) have [outside seating](attribute:outside-seating)?
+    - what is the [price range](attribute:price-range) of [Berlin Burrito Company](restaurant)?
+    - what is with [I due forni](restaurant)?
+     ...
+
+The above examples just show examples related to the restaurant domain.
+You should add examples for every object type that exists in your knowledge base.
+
+All user requests can be divided into two categories:
+(1) The user wants to obtain a list of objects of a specific type or (2) the user wants to know about a certain
+attribute of an object.
+The ``ActionQueryKnowledgeBase`` can handle both of those requests.
+Other requests, such as comparison between objects, are currently not supported.
+
+Another thing you may have noticed is, that we marked different kind of entities in the NLU data.
+If you want to use ``ActionQueryKnowledgeBase``, you need to specify the following entities:
+
+- ``object_type``: Whenever the user is talking about a specific object type from your knowledge base, the type should
+  be marked as entity in our NLU data. Use :ref:`entity_synonyms` to map, for example, "restaurants" to the correct
+  object type listed in the knowledge base, e.g. "restaurant".
+- ``mention``: If the user refers to an object via "the first one", "that one", or "it", you should mark those terms
+  as ``mention``. We also use :ref:`entity_synonyms` to map some of the mentions to symbols. More on that in
+  :ref:`resolve_mentions`.
+- ``attribute``: All attribute names defined in your knowledge base should be marked as ``attribute`` in the NLU data.
+  Again, use :ref:`entity_synonyms` to map variations of an attribute name to the one used in the knowledge base.
+
+Don't forget to add those entities to your domain file (as entities and slots).
```

Comment:
```suggestion
Remember to add those entities to your domain file (as entities and slots).
```

Reply:
do they need to be slots? should the be unfeaturized? let's include a snippet (possibly from the example)